includes/components/class-model.php

```diff
@@ -17,6 +17,20 @@
  */
 final class Model extends Component {
 
+	/**
+	 * Class constructor.
+	 *
+	 * @param array $args Component arguments.
+	 */
+	public function __construct( $args = [] ) {
+
+		// Sync attributes.
+		add_action( 'hivepress/v2/models/user/update', [ $this, 'update_user' ], 100, 2 );
```

Comment:
Please always let me know before the decision to change/extend the base API, components like this one, or Hook, Cache, etc. are base ones, and adding public helpers is extending the API.

If possible rewrite this to leave the helper only in this component, and move hooks/callbacks related to specific models to the corresponding components.

Reply:
Also, move this helper to the Attribute component, it's more related to the Attributes API. Please simplify the helper, it can accept the initial model object to sync from, and a ready query to sync to, this way we'll avoid adding another wrapper for the query API and the query logic will be fully editable (not just filters) where the sync helper is being called.

Reply:
It was my decision to put it to the Model component to keep all sync functionality in one place. Yes, we will move it to the Attribute component and check again the helper function to simplify it